### PR TITLE
Dont `__call` nor `__callStatic`

### PR DESCRIPTION
**Pull request by BackEndTea**
Following [this](https://twitter.com/BackEndTea/status/1093809127246258177) twitter conversation.

I'm partial to making the error text a refference to Carly Rae Jespens 'Call me maybe', but i couldn't think of anything.

### CODE REVIEW

File: src/Dont/DontCall.php [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+use Dont\Exception\NonCallableObject;
+use Dont\Exception\TypeError;
+
+trait DontCall
+{
+    /**
+     * @throws NonCallableObject
+     * @throws TypeError
+     */
+    final public function __call($name, $arguments)
```
Comment (asgrim) [inline src/Dont/DontCall.php:16]: No `__callStatic`?

File: src/Dont/DontCall.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dont;
+
+use Dont\Exception\NonCallableObject;
+use Dont\Exception\TypeError;
+
+trait DontCall
```
Comment (asgrim) [inline src/Dont/DontCall.php:10]: `README.md` needs updating to add this please

**Review by asgrim — CHANGES_REQUESTED** [review]

File: src/Dont/DontCall.php (hunk 1 above)
Comment (BackEndTea) [reply src/Dont/DontCall.php:16]: Added a callStatic one as well

**Review by Ocramius — APPROVED** [review]
Damn, this was just sitting around! :+1:

**Comment by BackEndTea** [comment]
Thank you @Ocramius & @asgrim

**Pull request merged**